### DISCUSSION TITLE
Struggling with wifi connection

### DISCUSSION THREAD

**Discussion by gjf2a** [iRobotEducation/create3_docs#390] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

G.4.5

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

/archangel

### Is the robot connected to a network? If so, what is the network type?

Yes; educational network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

I'm struggling to connect my Create3 robot to my wifi network. 

I'm at a college and I am trying to use their network. I tried my home network also, though, and I had similar problems.

When I navigate to the robot and select Connect, I enter the wifi and password. I then get this message:
* Connecting to 'HendrixIOT'...
* Successfully sent provision message to the robot

After this, the robot cycles a bit, it displays blue, it chimes, then it displays orange and gives a different chime. 

I don't get anything else from the robot, and, in particular, I can't seem to discover its IP address. I used `arp -a` from the terminal, and only my machine shows up - no others. (It's summer, not much is happening here, so no surprise at a lack of other devices.)

My original trouble was with getting a ROS2 connection working. I read through [this discussion](https://github.com/iRobotEducation/create3_docs/discussions/368), and it seems that to set up unicast I need to know the robot's IP address. The user in that situation was somehow able to see the IP address from 192.168.10.1, but I get disconnected from the robot when I do the setup, and when I reconnect, there is no IP address displayed.

Any advice would be welcome. Thanks!

**Comment by shamlian** [comment]
Just a couple quick things that might help: 

1) A video of the light ring and chimes could help us to identify what is going wrong. There's not an orange ring during the provisioning flow; could it be [red over yellow](https://iroboteducation.github.io/create3_docs/hw/face/#while-connecting-to-wi-fi)? That means something is wrong with the password.
2) If your adapter board is set to BLE, you should be able to access the robot using the [Python Web Playground](python.irobot.com); once you connect to your robot, you can run create3_ip_address.py to get your robot's IP address (if it's successfully connected to your network).

**Reply by gjf2a** [reply]
It was indeed red-over-yellow. After carefully retyping the wifi password, I used the Python Web Playground as you suggested, and as it turns out the robot is, in fact, connected to our wifi network. I then did the following:
* I verified the wifi connection by opening the IP address in my browser. It works fine - I get the screen, I can see the IP address, etc.
* I created a unicast_locator.xml file along the lines suggested [here](https://github.com/iRobotEducation/create3_docs/discussions/368#discussioncomment-5511704).
* I did, of course, edit in my robot's actual IP address.
* I ran `export FASTRTPS_DEFAULT_PROFILES_FILE=/home/gjf2a/unicast_locator.xml`
* I configured my `.bashrc` to set the RMW to `rmw_fast_rtps_cpp`, and I did the same on the robot side.
* I stopped the daemon and checked the ros2 topic list. 
* Still nothing.

What else should I try? I'm eager for suggestions.

Thanks!

**Reply by gjf2a** [reply]
At this point, I've figured out how to get a wired connection working with my Libre AML-S905X-CC board - I plan to run ROS 2 on that, then talk to the board via wifi, rather than controlling ROS 2 via my laptop.

**Reply by kwujciak** [reply]
Did the Libre work? Make sure if you're using a board instead of wifi, you factory reset your robot so it does not automatically connect to your wifi. If it automatically connects to wifi and your robot is USB connected to the board, that can cause some issues. A warning though - if you factory reset, this will remove all user configuration (firmware version will not change). 

In response to above, if it is also not working on your home network, it is likely a problem with the connection. It seems like you took all the right steps to unicast. What happens when you ping your IP in your VM terminal?

`ping <your IP>`

Make sure your domain ID, namespace, and [RMW](https://iroboteducation.github.io/create3_docs/setup/xml-config/) are consistent across all platforms (in the config page of the webserver and in your bash script). For example:

```
export FASTRTPS_DEFAULT_PROFILES_FILE='/home/{your user}/Documents/disable_multicast_example.xml'
export ROS_DOMAIN_ID=0
export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
```
If these are in your bash script, cross-compare with your config page. Make sure your IP is correct in your .xml and **hasn't changed** . Sometimes, IPs can change if they are not static. 

Then you can run these commands:

```
ros2 daemon stop
source ~/.bashrc
ros2 topic list
```

Did this work for wifi? Were you successful with the Libre?

**Reply by gjf2a** [reply]
So it turns out that the Libre microUSB power connection is not a data connection, so unlike the Raspberry Pi, that connection doesn't do the job.

For reasons I can't quite figure out, though, the ROS2 wifi connection is now working. And, in fact, when I believed it was the wired connection working, it was actually the wifi connection working the whole time!

The wifi performance is slow. It takes a while to "warm up" and get the nodes talking to each other. Once Raspberry Pis are available again, I think I will try buying one and using it instead of the Libre.
